File: leet_BS_611_valid_triangle.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        s3 = len(nums) - 1
        count = 0
        nums.sort()
        while s3 > 1:
            s1 = 0
            s2 = s3 - 1
            while s1 < s2:
                if nums[s1] + nums[s2] > nums[s3]:
                    count += s2 - s1
                    s2 -= 1
                else:
                    s1 += 1
            s3 -= 1
        return count

File: leet_BS_611_valid_triangle.cpp
class Solution {
public:
    int triangleNumber(vector<int>& nums)
    {
        int s1 = (int) nums.size();
        int s2, s3, count = 0;
        sort(nums.begin(), nums.end());
        while (--s1 > 1)
        {
            s2 = 0, s3 = s1 - 1;
            while (s2 < s3)
            {
                if (nums[s2] + nums[s3] > nums[s1])
                {
                    count += s3 - s2;
                    --s3;
                }
                else
                    ++s2;
            }
        }
        return count ;
    }
};
